cider/tests/nextgen/jitlib/JITLibTest.cpp

```diff
@@ -143,6 +146,43 @@ TEST_F(JITLibTests, LogicalOpTest) {
       false, true, true, [](JITValue& a, JITValue& b) { return !a; });
 }
 
+TEST_F(JITLibTests, TupleTest) {
+  LLVMJITModule module("Test Module");
+  JITTuple layer1(JITTypeTag::TUPLE, JITBackendTag::LLVMJIT);
+  JITTuple layer2(JITTypeTag::TUPLE, JITBackendTag::LLVMJIT);
+  JITTuple layer3(JITTypeTag::TUPLE, JITBackendTag::LLVMJIT);
+
+  JITFunctionPointer function1 = module.createJITFunction(JITFunctionDescriptor{
+      .function_name = "test_Tuple",
+      .ret_type = JITFunctionParam{.type = JITTypeTag::INT32},
+      .params_type = {JITFunctionParam{.name = "a", .type = JITTypeTag::INT32}}});
+
+  JITValuePointer x_in = function1->createVariable("x1", JITTypeTag::INT32);
+  JITValuePointer a_in = function1->getArgument(0);
+  JITValuePointer b_in = function1->createConstant(JITTypeTag::INT32, 876);
+
+  layer2.append<JITValue>(b_in.get());
+  layer1.append<JITValue>(x_in.get());
+  layer1.append<JITValue>(a_in.get());
+  //   layer1.insert<JITValue>(b_in.get(), 1);
+  layer1.insert<JITTuple>(&layer2, 1);
+  layer1.append<JITTuple>(&layer3);
+
+  auto& x_out = layer1.get_as<JITValue>(0);
+  auto& layer2_out = layer1.get_as<JITTuple>(1);
+  auto& a_out = layer1.get_as<JITValue>(2);
+  auto& b_out = layer2_out.get_as<JITValue>(0);
+  //   auto& b_out = layer1.get_as<JITValue>(1);
+
```

Comment:
if I pass layer1 to another function, how do I know what is in the JITTuple?

Reply:
op holds values information